cmd/snap/cmd_set_test.go

```diff
@@ -127,6 +127,22 @@ func (s *snapSetSuite) TestSnapSetFailParsingWithStrictJSON(c *check.C) {
 	c.Assert(err, check.ErrorMatches, "failed to parse JSON:.*")
 }
 
+func (s *snapSetSuite) TestSnapSetFailJsonKeysValidation(c *check.C) {
+	invalidJson := []string{
+		`key={"key0":"val0","key1":"val1","key2":{"key3":"val2","key4":"val3",` +
+			`"key5":{"INVALID":"val4","key7":"val5"}},"key8":{"key9":{"key10":"val6"}}}`,
+
+		`key={"key0":"val0","key1":"val1","key2":{"key3":"val2","key4":"val3",` +
+			`"key5":{"key6":"val4","key7":"val5"}},"key8":{"key9":{"key10":"val6"}},` +
+			`"key11":[{"key12":"val7"},{"KEY13":"val8"},{"key14":["val9","val10"]}]}`,
+	}
```

Comment:
Currently, both these test cases have invalid keys. Please include some test cases with all valid keys but invalid structure in other ways, all the ways you expect to catch.

For example, I can't tell at the moment whether some of the following should be valid or invalid:

```
key={"key0": ["a", "b", "c"]}       // value is arrays of primitives
key=[{"key1": "a"}, {"key2": "b"}]  // top-level value is array of objects
key="123"                           // top-level value is string
key=123                             // top-level value is int
key=[123, 456]                      // top-level value is array of ints
key={"key1": [123, "abc"]}          // value is array of mixed types
key={"key1": {"key2": "a", "key2": ["b"]}} // mixed primitive and array values
key={"key1": [[123, 456], [789]]}   // values is nested arrays
key=[[123, 456], [789]]             // top-level value is nested array
key=[123, [456, 789]]               // top-level value is array of int and array
key={"key": {"key": "value"}}       // duplicate nested key names
key={"key": "val", "key": "val"}    // duplicate key and value names in single object
key={123: 456}                      // int key in object
key={"key1": 123, 456: "val"}       // mixed key types
key={["k"]: ["v"]}                  // key is array
key={{"kk": "kv"}: {"vk": "vv"}}    // key is object
123="val"                           // int top-level key
```

Please create dedicated functions to test valid and invalid json. This test at the moment calls some parser-related function which eventually calls `Execute`, which calls the new `validateJsonKeys` function. I would prefer if these test functions were to call `validateJsonKeys` directly with the inputs you care about, or perhaps to call `Execute` directly with a `cmdSet` set up as minimally as possible.

If you want to call `validateJsonKeys` directly, you can put the following in `cmd/snap/export_test.go`:
```go
var ValidateJsonKeys = validateJsonKeys
```